src/ht.c

```diff
@@ -174,45 +179,30 @@ static bool internal_ht_insert_kv(SdbHt *ht, HtKv *kv, bool update) {
 			ht->table[bucket] = ls_newf ((SdbListFree)ht->freefn);
 		}
 		ls_prepend (ht->table[bucket], kv);
-#if INSERTORDER
-		ls_append (ht->list, kv);
-#endif
 		ht->count++;
-#if GROWABLE
 		// Check if we need to grow the table.
-		if (ht->count >= ht->load_factor * ht_primes_sizes[ht->prime_idx]) {
-			ht->prime_idx++;
+		if (ht->count >= LOAD_FACTOR * ht->size) {
 			internal_ht_grow (ht);
 		}
-#endif
 		return true;
 	}
 	return false;
 }
 
 static bool internal_ht_insert(SdbHt* ht, bool update, const char* key,
 				void* value) {
-	if (!ht || !key || !value) {
+	if (!ht) {
 		return false;
 	}
 	HtKv* kv = calloc (1, sizeof (HtKv));
 	if (kv) {
-		kv->key = ht->dupkey ((void *)key);
-		if (ht->dupvalue) {
-			kv->value = ht->dupvalue ((void *)value);
-		} else {
-			kv->value = (void *)value;
-		}
-		kv->key_len = ht->calcsizeK ((void *)kv->key);
-		if (ht->calcsizeV) {
-			kv->value_len = ht->calcsizeV ((void *)kv->value);
-		} else {
-			kv->value_len = 0;
-		}
+		kv->key = DUPKEY (ht, key);
```

Comment:
may be null and its not checked later

Reply:
true.. it was never checked, but I can do it now.